### PR TITLE
HGCal Reworked rechit scheme

### PR DESCRIPTION
**Pull request by lgray**
HGC Rechits are reworked such that uncalibrated Rechits are in MIPs.

This is preparation towards having an EM calibrated RecHit.

Also, the hexagon geometry is made default in unit tests from now on.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

File: RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h [hunk 1]
```diff
@@ -51,30 +60,37 @@ template<class C> class HGCalUncalibRecHitRecWeightsAlgo
...
+      LogDebug("HGCUncalibratedRecHit") << "ADC+: set the charge to: " << amplitude_ << ' ' << sample.data() 
                                         << ' ' << adcLSB_ << ' ' << std::endl;
     }
     
+    int thickness = 1;
+    if( ddd_ != nullptr ) {
+      HGCalDetId hid(dataFrame.id());
+      thickness = ddd_->waferTypeL(hid.wafer());
+    }
+    amplitude_ = amplitude_/fCPerMIP_[thickness-1];
```
Comment (cvuosalo) [inline RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h]: Please check if `fCPerMIP_[thickness-1]` is zero before dividing by it.

**Comment by cvuosalo** [comment]
The static analyzer complains about SimCalorimetry/HGCalSimProducers/src/HGCHEbackDigitizer.cc. It would be nice to get the issue fixed at some point. Lines 59-64 could be changed as follows.

```
59	  float totalMIPs(totalIniMIPs);
	
Value stored to 'totalMIPs' during its initialization is never read
60	      const float xtalk = (nTotalPE_-xTalk_*((float)nPixel))/(nTotalPE_-((float)nPixel));
61		  if( nTotalPE_ != nPixel && xtalk > 0. )
62		    totalMIPs = (nTotalPE_/nPEperMIP_)*vdt::fast_logf(xtalk);
63		  else
64		    totalMIPs = 0.f;
```
to
```
float totalMIPs(0.f), 
const float xtalk = 0.f,  peDiff = nTotalPE_ - (float) nPixel;
if (peDiff != 0.) {
  xtalk = (nTotalPE_-xTalk_*((float)nPixel)) / peDiff;
  if( xtalk > 0. && nPEperMIP_ != 0.)
   totalMIPs = (nTotalPE_/nPEperMIP_)*vdt::fast_logf(xtalk);
}
```
It could be the line `totalMIPs = 0.f;` is incorrect, so the initialization `float totalMIPs(totalIniMIPs);` would actually be correct.

**Comment by lgray** [comment]
@cvuosalo Thanks for the review. I am taking care of it now.

File: RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h (hunk 1 above)
Comment (lgray) [inline RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h]: I will check this in the constructor so it is guaranteed non-zero here.

**Comment by lgray** [comment]
@cmsbuild please test

I have addressed the code review.

**Comment by cvuosalo** [comment]
+1

For #13835 4f5269ca783ce77611868c9d1b3fa287977b2631

Revising HGCal rec hits. There should be no change in standard (non-HGCal) workflows.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-04-04-1100 show no significant differences, as expected.

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**